### PR TITLE
Update PerfTestRunner example and add network replay

### PR DESCRIPTION
**Pull request by trevor-e**
Notable changes:
- Updates `PerfTestRunner` to better mimic how we execute perf tests
- Adds `EMGURLProtocol` for others to add to their project to test network replay
- Adds a convenience function to set up our network replay
- Update our console output to be more prominent

Testing in my sample app, for the second iteration I see the recording happening:
```
Fetching request for https://jsonplaceholder.typicode.com/users
EMERGE: Starting loading for https://jsonplaceholder.typicode.com/users 
EMERGE: No matching path for https://jsonplaceholder.typicode.com/users 
EMERGE: Going to record for https://jsonplaceholder.typicode.com/users 
EMERGE: Starting loading for https://graph.facebook.com/v3.2/(null)/mobile_sdk_gk?fields=gatekeepers&format=json&include_headers=false&platform=ios&sdk=ios&sdk_version=4.44.1 
EMERGE: No matching path for https://graph.facebook.com/v3.2/(null)/mobile_sdk_gk?fields=gatekeepers&format=json&include_headers=false&platform=ios&sdk=ios&sdk_version=4.44.1 
EMERGE: Going to record for https://graph.facebook.com/v3.2/(null)/mobile_sdk_gk?fields=gatekeepers&format=json&include_headers=false&platform=ios&sdk=ios&sdk_version=4.44.1 
EMERGE: Finished url request https://jsonplaceholder.typicode.com/users 
Received response for https://jsonplaceholder.typicode.com/users
EMERGE: Finished url request https://graph.facebook.com/v3.2/(null)/mobile_sdk_gk?fields=gatekeepers&format=json&include_headers=false&platform=ios&sdk=ios&sdk_version=4.44.1 
```

in the third iteration I see the replaying:
```
Fetching request for https://jsonplaceholder.typicode.com/users
EMERGE: Starting loading for https://jsonplaceholder.typicode.com/users 
EMERGE: Trying file:///Users/telkins/Library/Developer/XCTestDevices/AA0604CC-C87E-4A1E-8412-6DC37AB3D1A4/data/Containers/Data/Application/23A26592-C472-432D-AE42-6CC25AF4A58A/Documents/emerge-cache/%3CTrevorSwiftUIUITests.EmergePerfTest:%200x600000ce82a0%3E/jsonplaceholder.typicode.com/users/2DA6E7CC-9959-43E4-9259-13FF587A4714 
EMERGE: Found matching request for https://jsonplaceholder.typicode.com/users, data size 5645, response <NSHTTPURLResponse: 0x600002014880> { URL: https://jsonplaceholder.typicode.com/users } { Status Code: 200, Headers {
    Age =     (
        28444
    );
    "Alt-Svc" =     (
        "h3=\":443\"; ma=86400, h3-29=\":443\"; ma=86400"
    );
    "Cache-Control" =     (
        "max-age=43200"
    );
    "Content-Encoding" =     (
        br
    );
    "Content-Type" =     (
        "application/json; charset=utf-8"
    );
    Date =     (
        "Thu, 22 Dec 2022 20:55:49 GMT"
    );
    Etag =     (
        "W/\"160d-1eMSsxeJRfnVLRBmYJSbCiJZ1qQ\""
    );
    Expires =     (
        "-1"
    );
    Pragma =     (
        "no-cache"
    );
    Priority =     (
        "u=3,i=?0"
    );
    Server =     (
        cloudflare
    );
    Vary =     (
        "Origin, Accept-Encoding"
    );
    Via =     (
        "1.1 vegur"
    );
    "access-control-allow-credentials" =     (
        true
    );
    "cf-cache-status" =     (
        HIT
    );
    "cf-ray" =     (
        "77dbc2b438c48c0b-EWR"
    );
    nel =     (
        "{\"success_fraction\":0,\"report_to\":\"cf-nel\",\"max_age\":604800}"
    );
    "report-to" =     (
        "{\"endpoints\":[{\"url\":\"https:\\/\\/a.nel.cloudflare.com\\/report\\/v3?s=Q2jEn5nNI69awPxcG4eYMD0QpBwN2gnUvFw8n7VLCpfcQJKUwxDNEcpQc1Lvxg9Huz8UY8ajlXJnVKFtBOsPYZ7qsB4CFyOh1IfZrRV40BNVYEnviAsJi04q0azhyivw4k0gyzlPmqJq26Y0x9nf\"}],\"group\":\"cf-nel\",\"max_age\":604800}"
    );
    "x-content-type-options" =     (
        nosniff
    );
    "x-powered-by" =     (
        Express
    );
    "x-ratelimit-limit" =     (
        1000
    );
    "x-ratelimit-remaining" =     (
        997
    );
    "x-ratelimit-reset" =     (
        1640243547
    );
} } 
Received response for https://jsonplaceholder.typicode.com/users
```

### CODE REVIEW

File: EMGLog.m [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    dispatch_once(&queueCreationGuard, ^{
+        uuid = [NSUUID new];
+    });
+    
+    va_list args;
+    va_start(args, format);
+    NSString *input = [[NSString alloc] initWithFormat:format arguments:args];
+    va_end(args);
+    NSLog(@"### EMG %@ %@", uuid, input);
+    printf("EMERGE: %s \n", [input cStringUsingEncoding:NSUTF8StringEncoding]);
```
Comment (trevor-e) [inline EMGLog.m:16]: I added an extra `printf`, for some reason `NSLog` wasn't reporting to the console logs when I was testing my sample app.

File: EMGURLProtocol.m [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+//  PerfTesting
+//
+//  Created by Noah Martin on 8/4/22.
+//
+
+#import "EMGURLProtocol.h"
+#import "EMGLog.h"
+#import "EMGTuple.h"
+
+@interface EMGCacheEntry : NSObject <NSSecureCoding>
```
Comment (trevor-e) [inline EMGURLProtocol.m:12]: I updated this to include the dictionary access crash fix.

File: NetworkReplay/EMGLog.m [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#import <Foundation/Foundation.h>
+#import "EMGLog.h"
+
+void EMGLog(NSString *format, ...) {
+    static dispatch_once_t queueCreationGuard;
+    static NSUUID *uuid;
+    dispatch_once(&queueCreationGuard, ^{
+        uuid = [NSUUID new];
```
Comment (trevor-e) [inline NetworkReplay/EMGLog.m:8]: I added an extra printf, for some reason NSLog wasn't reporting to the console logs when I was testing my sample app.

File: NetworkReplay/EMGLog.m [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    });
+    
+    va_list args;
+    va_start(args, format);
+    NSString *input = [[NSString alloc] initWithFormat:format arguments:args];
+    va_end(args);
+    NSLog(@"### EMG %@ %@", uuid, input);
+    printf("EMERGE: %s \n", [input cStringUsingEncoding:NSUTF8StringEncoding]);
```
Comment (trevor-e) [inline NetworkReplay/EMGLog.m:16]: I added an extra printf, for some reason NSLog wasn't reporting to the console logs when I was testing my sample app.

File: NetworkReplay/EMGURLProtocol.m [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+}
+
++ (BOOL)supportsSecureCoding
+{
+    return YES;
+}
+
+@end
+
+@implementation EMGURLProtocol
```
Comment (trevor-e) [inline NetworkReplay/EMGURLProtocol.m:60]: I updated this to include your latest dictionary access crash fix.

**Review by noahsmartin — APPROVED** [review]

**Pull request merged**